glmnet/tests/test_linear.py

```diff
@@ -176,6 +176,11 @@ def test_random_state_cv(self):
         print(dir(m.cv))
         assert m.cv.random_state == random_state
 
+    def test_max_features(self):
+        x, y = self.inputs[0]
+        m = ElasticNet(n_splits=3, random_state=42)
+        m = m.fit(x, y, max_features=100)
+        self.check_r2_score(y, m.predict(x), 0.90)
```

Comment:
thanks for adding a test! However, I think this may not actually test that `max_features` is observed. Perhaps instead of testing the R^2 score, we can check that the number of nonzero elements in the coefficient vector is bounded above by `max_features`.